src/bindings/js/node/src/napi_arg.cpp

```diff
@@ -0,0 +1,190 @@
+// Copyright (C) 2018-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+
+#include "node/include/napi_arg.hpp"
+
+namespace NapiTypename {
+static const char UNDEFINED_STR[] = "Undefined";
+static const char NULL_STR[] = "Null";
+static const char BOOLEAN_STR[] = "Boolean";
+static const char NUMBER_STR[] = "Number";
+static const char STIRNG_STR[] = "String";
+static const char SYMBOL_STR[] = "Symbol";
+static const char OBJECT_STR[] = "Object";
+static const char FUNCTION_STR[] = "Function";
+static const char EXTERNAL_STR[] = "External";
+static const char BIGINT_STR[] = "BigInt";
+static const char UNKNOWN_STR[] = "Unknown";
+}  // namespace NapiTypename
+namespace NapiArg {
+const char* get_type_name(napi_valuetype type) {
+    switch (type) {
+    case napi_undefined:
+        return NapiTypename::UNDEFINED_STR;
+
+    case napi_null:
+        return NapiTypename::NULL_STR;
+
+    case napi_boolean:
+        return NapiTypename::BOOLEAN_STR;
+
+    case napi_number:
+        return NapiTypename::NUMBER_STR;
+
+    case napi_string:
+        return NapiTypename::STIRNG_STR;
+
+    case napi_symbol:
+        return NapiTypename::SYMBOL_STR;
+
+    case napi_object:
+        return NapiTypename::OBJECT_STR;
+
+    case napi_function:
+        return NapiTypename::FUNCTION_STR;
+
+    case napi_external:
+        return NapiTypename::EXTERNAL_STR;
+
+    case napi_bigint:
+        return NapiTypename::BIGINT_STR;
+
+    default:
+        return NapiTypename::UNKNOWN_STR;
+    }
+}
+
+std::string create_error_message(const std::string& key, const char* expected, const char* real) {
+    return "Argument #" + key + " has type '" + std::string(real) + "', expected '" + std::string(expected) + "'";
+}
+
+std::string create_error_message(const std::string& key, napi_valuetype expected_type, napi_valuetype real_type) {
+    return create_error_message(key, get_type_name(expected_type), get_type_name(real_type));
+}
+
+std::string create_error_message(const std::string& key, const char* expected, const napi_valuetype& real_type) {
+    return create_error_message(key, expected, get_type_name(real_type));
+}
+
+void check_type(const napi_valuetype expected_type, const std::string& key, const Napi::Value& value) {
+    napi_valuetype real_type = value.Type();
```

Comment:
If I understand correctly `check_type` cannot check if the value is a `Tensor`.
In file _src/bindings/js/node/include/type_validation.hpp_ there is a function that perform this check: `bool is_tensor(const Napi::Env& env, const Napi::Value& value)`.